libunwind/src/DwarfInstructions.hpp

```diff
@@ -484,23 +412,14 @@ int DwarfInstructions<A, R>::stepWithDwarf(A &addressSpace, pc_t pc,
 #endif
 
 #if defined(__CHERI_PURE_CAPABILITY__) && defined(_LIBUNWIND_CHERI_C18N_SUPPORT)
```

Comment:
This whole block should probably be in UnwindCursor; that's where sigreturn gets handled, for example. It's not specific to DWARF, it applies equally to any of the other unwind formats.